linalg/densemat.cpp

```diff
@@ -436,6 +455,43 @@ double DenseMatrix::Det() const
             d[3] * (d[2] * d[7] - d[1] * d[8]) +
             d[6] * (d[1] * d[5] - d[2] * d[4]);
       }
+      case 4:
+      {
+         const double *d = data;
+         return
+            d[ 0] * (d[ 5] * (d[10] * d[15] - d[11] * d[14]) -
```

Comment:
Let's not use the recursive algorithm - its complexity for `n x n` matrix is worse than `n!`. The class `LUFactors` should be a better option for `n>3`, if my counting is right.